types.go

```diff
@@ -150,3 +150,21 @@ type Capabilities interface {
 	// Tagging returns whether the reporter has the capability for tagged metrics.
 	Tagging() bool
 }
+
+// SuccessFilter determines whether an error should be considered success
+type SuccessFilter func(err error) bool
```

Comment:
nit: Can we standardize the  naming just a little and name this `SuccessFilterFn` to match the suffix of `ExecFn`?

Reply:
Sounds good 👍